bench: refactor random number generation

saurabhraghuvanshii opened the pull request:
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactors random number generation in JS benchmark in `math/base/special/besselj*`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
bench: refactor random number generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
@saurabhraghuvanshii Thanks for working on this. If you're already covering the functions from the top, I'll start from the bottom to avoid duplicate PRs. 😄

saurabhraghuvanshii commented:
Yes I am covering from top , thanks

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/besselj0/benchmark/c/native/benchmark.c:
```diff
@@ -90,16 +90,19 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 100000.0 * rand_double() ) - 0.0;
```
```suggestion
		x[ i ] = 100000.0 * rand_double();
```

saurabhraghuvanshii replied:
so many file have this so can I remove this type things that file too?

anandkaranubc replied:
Let's apply this change to the packages we've already refactored. For the remaining packages, we can implement this change along with other refactoring tasks.

saurabhraghuvanshii replied:
Ok thanks

anandkaranubc replied:
But yes, this type of change is desired in all the upcoming PRs.

saurabhraghuvanshii replied:
Today I do refactor for other files also , I have my end sem exam therefore, I am busy

anandkaranubc replied:
No rush! I don't think this is a high-priority change....

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/besselj1/benchmark/c/native/benchmark.c:
```diff
@@ -90,16 +90,19 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 100000.0 * rand_double() ) - 0.0;
```
```suggestion
		x[ i ] = 100000.0 * rand_double();
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely0/benchmark/c/native/benchmark.c:
```diff
@@ -90,16 +90,19 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 100000.0 * rand_double() ) - 0.0;
```
```suggestion
		x[ i ] = 100000.0 * rand_double();
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/beta/benchmark/c/cephes/benchmark.c:
```diff
@@ -94,18 +94,21 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 1000.0*rand_double() ) + 0.0;
+		y[ i ]= ( 1000.0*rand_double() ) + 0.0;
```
```suggestion
		x[ i ] = 1000.0 * rand_double();
		y[ i ]= 1000.0 * rand_double();
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/beta/benchmark/c/native/benchmark.c:
```diff
@@ -90,18 +90,21 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( ( rand_double() * 1000.0 ) + 0.0 );
+		y[ i ]= ( ( rand_double() * 1000.0 ) + 0.0 );
```
```suggestion
		x[ i ] = rand_double() * 1000.0;
		y[ i ]= rand_double() * 1000.0;
```

anandkaranubc reviewed (changes requested):
Just some minor changes :)

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into random-refactor

saurabhraghuvanshii committed:
chore: address comments change

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

anandkaranubc commented:
@kgryte This should be ready to merge.
